### PR TITLE
Feature #1645 vepcasl phoenix update

### LINKED ISSUE #1645: Update VE-PCASL Phoenix to multi-PLD
### Description

Update the VE-PCASL Phoenix flavor to the multi-PLD sequence. Which mainly means adding more PLD fields

### Tasks

- [x] Fix a bug, labeling duration is field 9, not 11
- [x] Add further PLD fields
- [x] Add 3D version
- [x] String parsing is messed up and %\to is parsed as %o in protocol name
- [x] Automatically match ASLcontext when PLD vector only covers PLDs read automatically and not covers M0
- [x] Update the reference of the older flavor
- [x] Check VE flavors
- [x] Make sure the non-Phoenix flavors work
- [x] Add new flavor - PCASL2D,PCASL3D
- [x] Add to the flavor list and mark
- [x] Find all other similar flavors and update their studyPars and Phoenix-flavor database.
- [x] Fix flavor differences

### Release notes

Parse Phoenix protocol of multi-PLD VEPCASL protocol.

### PR DESCRIPTION
**Pull request by jan-petr**
### Linked issue

Close #1645

### CODE REVIEW

**Comment by jan-petr** [comment]
Flavor test almost good ;)


COMPARISON TABLE:]
                    flavor                     dataset                name                                                 message                                   
    _______________________________________    ________    __________________________    ____________________________________________________________________________

    {'GE_PCASL_3Dspiral_DV25.0_1.5T_1'    }    {'Both'}    {'Different file content'}    {'Different file content: /sub-D21002/perf/sub-D21002_asl.json'            }
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /sub-101/ses-1/perf/sub-101_ses-1_run-5_asl.json'}
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /sub-101/ses-1/perf/sub-101_ses-1_run-6_asl.json'}
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /sub-101/ses-3/perf/sub-101_ses-3_run-5_asl.json'}
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /sub-101/ses-
    {'Siemens_PCASL_multiSequence_VE11C_2'}    {'Both'}    {'Different file content'}    {'Different file content: /sub-sub1/per
    {'GE_PCASL_3Dspiral_DV25.0_1.5T_1'    }    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_1'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s
    {'Siemens_PCASL_multiSequence_VE11C_2'}    {'Both'}    {'Different file content'}    {'Different file content: /ExploreASL/s

LOGGING TABLE:]


ans =

File: Modules/Module_Import/NII2BIDS/xASL_bids_PhoenixProtocolAnalyzer.m [hunk 1]
```diff
@@ -124,18 +153,38 @@
 
 	%% 3. Reading sequence VEPCASL
 	if ~bSequenceIdentified && ~isempty(regexp(sourcePar.tSequenceFileName,'VEPCASL', 'once'))
-		if ~isempty(regexp(sourcePar.tSequenceFileName,'ep2d_VEPCASL', 'once'))
+		% Basic check that the PCASL tag is checked
```
Comment (HenkMutsaerts) [inline Modules/Module_Import/NII2BIDS/xASL_bids_PhoenixProtocolAnalyzer.m]: Unclear comment. A box can be checked. A tag cannot be checked in the same way (though you can check if it is empty or on).

So do you mean:

"is verified"?

and what is a "basic check", is there also an advanced check? Perhaps be specific or not write anything :)

File: Modules/Module_Import/NII2BIDS/xASL_bids_PhoenixProtocolAnalyzer.m [hunk 2]
```diff
@@ -124,18 +153,38 @@
...
+		if isempty(sourcePar.sWipMemBlockalFree0) || sourcePar.sWipMemBlockalFree0 ~= 1
+			error('VEPCASL sequence without PCASL activated');
+		end
+		% 2D-EPI VEPCASL
+		if ~isempty(regexp(sourcePar.tSequenceFileName,'o_ep2d_VEPCASL', 'once'))
 			% Basic check
 			if ~isempty(sourcePar.sWipMemBlockalFree9) &&  ~isempty(sourcePar.sWipMemBlockalFree11) 
 				% All basic passes checked, we can proceed to parameter parsing
```
Comment (HenkMutsaerts) [inline Modules/Module_Import/NII2BIDS/xASL_bids_PhoenixProtocolAnalyzer.m]: What is a "pass-check"? What is a basic pass? :)

**Review by HenkMutsaerts — APPROVED** [review]
Some remarks about unclear comments but you can also ignore them :)

File: Modules/Module_Import/NII2BIDS/xASL_bids_PhoenixProtocolAnalyzer.m (hunk 1 above)
Comment (jan-petr) [reply Modules/Module_Import/NII2BIDS/xASL_bids_PhoenixProtocolAnalyzer.m]: Good point. Fixed.

File: Modules/Module_Import/NII2BIDS/xASL_bids_PhoenixProtocolAnalyzer.m (hunk 2 above)
Comment (jan-petr) [reply Modules/Module_Import/NII2BIDS/xASL_bids_PhoenixProtocolAnalyzer.m]: Good point. Fixed

**Pull request merged**